### ISSUE TITLE
inotify not registering files

### ISSUE THREAD

**Issue by obi1** [rakshasa/rtorrent#936]
about 20-30% of the files I download are being missed by inotify watch. If I redownload or touch the file it start downloading just fine.

entry in my rc file:
```
method.insert = load_other, simple|private, "load.start=(argument.0),d.directory.set=/home/rtorrent/data/downloads/other,d.custom1.set=other"
directory.watch.added = "/home/rtorrent/torrents/other/","load_other"
```
edit:  rTorrent 0.9.7/0.13.7

**Comment by chros73** [comment]
Normally it should work like this:
```ini
directory.watch.added = "/home/rtorrent/torrents/other/", load.start
```

I haven't tried myself the [wiki example](https://github.com/rakshasa/rtorrent/wiki/TORRENT-Watch-directories#inotify) though.

**Comment by obi1** [comment]
Maybe this should be changed to a libtorrent issue.

I did some testing and the problem occurs in combination with browsers like chromium/firefox using temporary download files in the same directory.

Unlike the older watch implementation that scanned the folder for all *.torrent files (if that is what was set in the config file), inotify reports every file. So sometimes the rename occurs a bit faster then the insert in rtorrent/libtorrent.

A dirty hack for this that works for me is checking the file extensions by modifying libtorrent

> torrent/utils/directory_events.cc line 157:

```
if (itr != m_wd_list.end()) 
        itr->slot(itr->path + event->name);
```

for 
```
if (itr != m_wd_list.end()) {
      std::string sname(event->name);
      if((sname.substr(sname.find_last_of(".") ) == ".torrent"))
        itr->slot(itr->path + event->name);
    }
```

**Comment by rakshasa** [comment]
Could you make that a pull request?

**Comment by obi1** [comment]
Fixed by rakshasa/libtorrent#203
